src/Common/src/Interop/Unix/System.Native/Interop.ForkAndExecProcess.cs

```diff
@@ -17,6 +17,20 @@ internal static partial class Sys
             bool redirectStdin, bool redirectStdout, bool redirectStderr,
             out int lpChildPid, out int stdinFd, out int stdoutFd, out int stderrFd, bool shouldThrow = true)
         {
+            if (string.IsNullOrEmpty(filename))
+            {
+                if (shouldThrow)
```

Comment:
> if (shouldThrow) [](start = 15, length = 17)

Should we throw anyway even shouldThrow is true?

Reply:
no, because when UseShellExecute is set, we could skip throwing so we can retry with another set of filename and args